packages/common/src/converter/search-attribute-payload-converter.ts

```diff
@@ -0,0 +1,58 @@
+import { ValueError } from '@temporalio/internal-workflow-common';
+import { PayloadConverter } from './payload-converter';
+import { JsonPayloadConverter } from './json-payload-converter';
+import { Payload, str } from './types';
+
+const jsonConverter = new JsonPayloadConverter();
+
+/**
+ * Converts Search Attribute values using JsonPayloadConverter
+ */
+export class SearchAttributePayloadConverter implements PayloadConverter {
+  public toPayload(values: unknown): Payload | undefined {
+    if (!(values instanceof Array)) {
+      throw new ValueError(`SearchAttribute value must be an array`);
+    }
+
+    const firstValue = values[0];
+    const firstType = typeof firstValue;
+    if (firstType === 'object') {
+      for (const value of values) {
+        if (!(value instanceof Date)) {
+          throw new ValueError(
+            `SearchAttribute values must arrays of strings, numbers, booleans, or Dates. This value ${value} is a ${typeof value}.`
+          );
+        }
+      }
+    } else {
+      for (const value of values) {
```

Comment:
value could be undefined or bigint, we should should prevent that, let's have an explicit whitelist of types (string, number, boolean).